src/main/scala/is/hail/methods/LDMatrix.scala

```diff
@@ -1,53 +1,134 @@
 package is.hail.methods
 
 import is.hail.utils._
-import is.hail.stats.ToNormalizedIndexedRowMatrix
 import is.hail.variant.{Variant, VariantDataset}
-import org.apache.spark.mllib.linalg.Matrix
-import org.apache.spark.mllib.linalg.distributed.{IndexedRow, IndexedRowMatrix}
-import org.apache.spark.storage.StorageLevel
+import org.apache.spark.Partitioner
+import org.apache.spark.mllib.linalg.{DenseMatrix, Matrix}
+import org.apache.spark.mllib.linalg.distributed.{BlockMatrix, IndexedRowMatrix}
+import org.apache.spark.rdd.RDD
+
+import scala.reflect.classTag
 
 
 object LDMatrix {
 
   /**
     * Computes the LD matrix for the given VDS.
     * @param vds VDS on which to compute Pearson correlation between pairs of variants.
+    * @param blockSize Size of groups to make in LD matrix computation.
     * @return An LDMatrix.
     */
-  def apply(vds : VariantDataset): LDMatrix = {
+  def apply(vds: VariantDataset, blockSize: Int): LDMatrix = {
     val nSamples = vds.nSamples
+
+    val bitPackedOpts = vds.rdd.map { case (_, (_, gs)) => LDPrune.toBitPackedVector(gs.hardCallIterator, nSamples)}
+    val filterArray = bitPackedOpts.map(!_.isEmpty).collect()
     val originalVariants = vds.variants.collect()
-    assert(originalVariants.isSorted, "Array of variants is not sorted. This is a bug.")
+    val variantsKeptArray = originalVariants.zip(filterArray).filter(_._2).map(_._1)
+    val nVariantsDropped = originalVariants.length - variantsKeptArray.length
+
+    val bpvs = bitPackedOpts.flatMap(x => x).zipWithIndex().map { case (a, b) => (b, a) }
```

Comment:
If I use "zip" to avoid doing the map flip, it'll require creating a local range then distributing it with parallelize then zipping it. Probably altogether worse than flipping. I'll defer this to a later map to avoid the intermediate.